fundingmanager.go

```diff
@@ -689,21 +689,36 @@ func (f *fundingManager) CancelPeerReservations(nodePub [33]byte) {
 }
 
 // failFundingFlow will fail the active funding flow with the target peer,
-// identified by its unique temporary channel ID. This method is send an error
-// to the remote peer, and also remove the reservation from our set of pending
-// reservations.
+// identified by its unique temporary channel ID. This method will send an
+// error to the remote peer, and also remove the reservation from our set of
+// pending reservations.
 //
 // TODO(roasbeef): if peer disconnects, and haven't yet broadcast funding
 // transaction, then all reservations should be cleared.
 func (f *fundingManager) failFundingFlow(peer *btcec.PublicKey,
-	tempChanID [32]byte, msg []byte) {
+	tempChanID [32]byte, fundingErr error) {
+
+	// We only send the exact error if it is part of out whitelisted set of
+	// errors (lnwire.ErrorCode or lnwallet.ReservationError).
+	var msg lnwire.ErrorData
+	switch fundingErr.(type) {
+
+	// Let the actual error message be sent to the remote.
+	case lnwire.ErrorCode, lnwallet.ReservationError:
+		msg = []byte(fundingErr.Error())
+
+	// We just send a generic error.
+	default:
+		msg = []byte("error")
```

Comment:
Maybe add a _bit_ more detail, something like: "failing funding flow due to internal error. Possibly we should also log the error in this case, to ease debugging in the future.

Reply:
Fixed.

The original error is logged below.